FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1494 @@
+    xContext.subscriptionCount = 1;
+    LogInfo( ( "Adding unsubscribe operation\n" ) );
+    prvAddCommandToQueue( &xCommand );
+    LogInfo( ( "Starting wait on operation\n" ) );
+
+    while( ( ulNotification & UNSUBSCRIBE_COMPLETE_BIT ) != UNSUBSCRIBE_COMPLETE_BIT )
+    {
+        LogInfo( ( "Waiting for unsubscribe operation to complete." ) );
+        xTaskNotifyWait( 0, UNSUBSCRIBE_COMPLETE_BIT, &ulNotification, DEMO_TICKS_TO_WAIT );
+
+        if( ++usWaitCounter > MAX_WAIT_ITERATIONS )
+        {
+            break;
+        }
+    }
+
+    LogInfo( ( "Operation wait complete.\n" ) );
+
+    /* Create command to stop command loop. */
+    LogInfo( ( "Beginning command queue termination." ) );
+    prvCreateCommand( TERMINATE, NULL, NULL, &xCommand );
+    prvAddCommandToQueue( &xCommand );
+
+    /* Notify main task this task can be deleted. */
+    xTaskNotify( xMainTask, SUBSCRIBE_TASK_COMPLETE_BIT, eSetBits );
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvMQTTDemoTask( void * pvParameters )
+{
+    NetworkContext_t xNetworkContext = { 0 };
+    PlaintextTransportStatus_t xNetworkStatus;
+    BaseType_t xResult;
+    uint32_t ulNotification = 0;
+    Command_t xCommand;
+
+    ( void ) pvParameters;
+
+    ulGlobalEntryTimeMs = prvGetTimeMs();
+
+    xCommandQueue = xQueueCreate( COMMAND_QUEUE_SIZE, sizeof( Command_t ) );
+    xSubscriberResponseQueue = xQueueCreate( PUBLISH_QUEUE_SIZE, sizeof( PublishElement_t ) );
+    /* Publish task doesn't receive anything in this demo, so it doesn't need a large queue. */
+    xPublisherResponseQueue = xQueueCreate( 1, sizeof( PublishElement_t ) );
+    /* In this demo, send publishes on non-subscribed topics to this queue. */
+    xDefaultResponseQueue = xPublisherResponseQueue;
+
+    memset( ( void * ) pxPendingAcks, 0x00, PENDING_ACKS_MAX_SIZE * sizeof( AckInfo_t ) );
+    memset( ( void * ) pxSubscriptions, 0x00, SUBSCRIPTIONS_MAX_COUNT * sizeof( SubscriptionElement_t ) );
+
+    /* Create inital process loop command. */
+    prvCreateCommand( PROCESSLOOP, NULL, NULL, &xCommand );
+    prvAddCommandToQueue( &xCommand );
+
+    LogInfo( ( "Creating a TCP connection to %s.\r\n", BROKER_ENDPOINT ) );
+
+    /* TODO: Use TLS to connect to the broker. */
+    xNetworkStatus = Plaintext_FreeRTOS_Connect( &xNetworkContext,
+                                                 BROKER_ENDPOINT,
+                                                 BROKER_PORT,
+                                                 TRANSPORT_SEND_RECV_TIMEOUT_MS,
+                                                 TRANSPORT_SEND_RECV_TIMEOUT_MS );
+    configASSERT( xNetworkStatus == PLAINTEXT_TRANSPORT_SUCCESS );
+    ( void ) prvMQTTConnect( &globalMqttContext, &xNetworkContext, true );
+    configASSERT( globalMqttContext.connectStatus = MQTTConnected );
+
+    xResult = xTaskCreate( prvSubscribeTask, "Subscriber", democonfigDEMO_STACKSIZE, NULL, tskIDLE_PRIORITY, &xSubscribeTask );
```

Comment:
As a general comment, I think some more comments can be added to the demo explaining each steps to improve the readability of the demo. Please refer to other mqtt demos in this repo. However, it can be added as a follow up PR as well.